model-optimizer/mo/ops/strided_slice.py

```diff
@@ -97,55 +54,86 @@ def convert(attr):
+        begin_mask = extend_mask(node.begin_mask, 1)  # differs from the case when we unroll ellipsis
+        end_mask = extend_mask(node.end_mask, 1)  # when we unroll ellipsis zeros should be inserted
+        # no need to extend ellipsis
+
+        # unroll ellipsis
+        if np.any(node.ellipsis_mask):
+            i = np.nonzero(node.ellipsis_mask)
+            assert len(i[0]) == 1, 'only one nonzero value in ellipsis_mask is allowed'
+            ellipsis_start = i[0][0]
+            # since we don't expect values in begin, end values and take all range of values along ellipsis_start axis
+            begin_mask[ellipsis_start] = 0
+            end_mask[ellipsis_start] = 0
+
+            num = input_rank - len(begin) + np.count_nonzero(node.new_axis_mask[ellipsis_start:])
+            unroll_ellipsis = lambda mask, val=0: np.insert(mask, ellipsis_start + 1, [val] * num).astype(int)
+
+            new_axis_mask = unroll_ellipsis(new_axis_mask)
+            shrink_axis_mask = unroll_ellipsis(shrink_axis_mask)
+            begin_mask, end_mask = unroll_ellipsis(begin_mask), unroll_ellipsis(end_mask)
+            begin, end, strides = unroll_ellipsis(begin), unroll_ellipsis(end), unroll_ellipsis(strides, 1)
+
+        # from now slices are without ellipsis
+        slice_rank = len(begin)
+        slices = [[]] * slice_rank
+        in_idx = 0  # index along input tensor shapes, note that input_rank not necessary is equal to slice_rank
+        for i in range(slice_rank):
+            if new_axis_mask[i]:
+                slices[i] = np.newaxis
+            elif shrink_axis_mask[i]:
+                slices[i] = int(begin[i])
+                if slices[i] < 0:
+                    slices[i] += int(shape[in_idx])
+            else:
+                start, stop = begin[i], end[i]
+                if not begin_mask[i]:  # if begin, and end are not specified take whole range
+                    start = 0 if strides[i] > 0 else -1
+                if not end_mask[i]:
+                    stop = shape[in_idx] if strides[i] > 0 else -shape[in_idx] - 1
+                slices[i] = slice(start, stop, strides[i])
+            in_idx += 1 if not new_axis_mask[i] else 0
+
+        if value is not None:
+            node.out_port(0).data.set_value(value[tuple(slices)])
+        else:
+            node.out_port(0).data.set_shape(get_shape_from_slice(shape, slices))
+
+        # normalize slices attr which is used by ConvertGroupedStridedSlice
+        in_idx = 0
+        for i in range(slice_rank):
+            if new_axis_mask[i]:
+                slices[i] = slice(0, 1, 1)
+            elif shrink_axis_mask[i]:
+                slices[i] = slice(slices[i], slices[i] + 1, strides[i])
+            if not new_axis_mask[i]:
+                slices[i] = slice(*slices[i].indices(shape[in_idx]))  # will convert negative indices
+                in_idx += 1
+        node['slices'] = np.array(slices)
```

Comment:
do we really need to convert this to numpy array?

Reply:
I leaved it as it was before me because changes potentially can break ConvertGroupedStridedSlice